### PR TITLE
fix: Remove room from UI when another agent takes it

### PR DESCRIPTION
**Pull request by KevLehman**
## Issue(s)

[OC-680](https://rocketchat.atlassian.net/browse/OC-680)
OC-873
[OC-1025](https://rocketchat.atlassian.net/browse/OC-1025)

## Further comments

[OC-680]: https://rocketchat.atlassian.net/browse/OC-680?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by aleksandernsilva — APPROVED** [review]

File: apps/meteor/server/models/raw/Users.js [hunk 1]
```diff
@@ -2875,4 +2875,27 @@ export class UsersRaw extends BaseRaw {
...
+		}
+
+		return this.find(
+			{
+				roles: {
+					$all: ['livechat-agent'],
+					$nin: ['livechat-manager', 'livechat-monitor'],
+				},
+				statusLivechat: 'available',
+				status: 'online',
```
Comment (murtaza98) [inline apps/meteor/server/models/raw/Users.js:2895]: Shouldn't we also respect `Livechat_enabled_when_agent_idle` setting over here?

Comment (KevLehman) [reply apps/meteor/server/models/raw/Users.js:2895]: Hmmmm, we can remove this filter as all we need is the "available" thing

**Review by murtaza98 — DISMISSED** [review]

**Review by aleksandernsilva — APPROVED** [review]

**Review by murtaza98 — APPROVED** [review]

File: apps/meteor/app/livechat/server/hooks/notifyInquiryTaken.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+			void api.broadcast('omnichannel.events', user._id, {
+				event: 'InquiryTaken',
+				data: {
+					inquiryId: inquiry._id,
+					roomId: room._id,
+					agentId: servedBy._id,
+				},
+			});
+		}
+	},
```
Comment (ggazzo) [inline apps/meteor/app/livechat/server/hooks/notifyInquiryTaken.ts:47]: I think this should/could be done inside the `listener.modules`

File: apps/meteor/server/modules/listeners/listeners.module.ts [hunk 3]
```diff
@@ -344,6 +344,10 @@ export class ListenersModule {
 			notifications.notifyUserInThisInstance(userId, 'voip.events', data);
 		});
 
+		service.onEvent('omnichannel.events', (userId, data): void => {
+			notifications.notifyUserInThisInstance(userId, 'omnichannel.events', data);
+		});
```
Comment (ggazzo) [inline apps/meteor/server/modules/listeners/listeners.module.ts:349]: I'm not sure informing every agent online (by user) about on inquiry being taken this is most performant approach.

If I'm not mistaken we already have the `roomData` that will be triggered and only who open the channel will receive

File: apps/meteor/app/livechat/server/hooks/notifyInquiryTaken.ts (hunk 2 above)
Comment (KevLehman) [reply apps/meteor/app/livechat/server/hooks/notifyInquiryTaken.ts:47]: And i was trying to not to add custom logic inside that file 😬

File: apps/meteor/server/modules/listeners/listeners.module.ts (hunk 3 above)
Comment (sampaiodiego) [reply apps/meteor/server/modules/listeners/listeners.module.ts:349]: agreed.. that would be much better to have some logic applied on on client side when data is received through the `room-data` stream

**Pull request closed without merging**